extensions/amp-google-assistant-assistjs/0.1/assistjs-frame-service.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview The shared service used by all custom elements to talk to Assistant services. It loads an iframe that exports
+ * endpoints to handle requests from custom elements.
+ */
+
+import 'regenerator-runtime/runtime';
+import {Services} from '../../../src/services';
+import {addAttributesToElement} from '../../../src/dom';
+import {toggle} from '../../../src/style';
+
+export class AssistjsFrameService {
+  /**
+   * @param {!../../../src/service/ampdoc-impl.AmpDoc} ampdoc
+   */
+  constructor(ampdoc) {
+    /** @private @const {!../../../src/service/ampdoc-impl.AmpDoc} */
+    this.ampDoc_ = ampdoc;
+
+    /** @private {?AssistjsConfigService} */
+    this.configService_ = null;
+
+    /** Create Assistant iframe and append it to the main AMP document. */
+    this.createAssistantIframe_();
+  }
+
+  /** @private */
+  createAssistantIframe_() {
+    this.configService_ = Services.assistjsConfigServiceForDoc(this.ampDoc_);
```

Comment:
`assistjsConfigServiceForDoc` returns a `Promise<ConfigService>`. A Promise means that it only has the `then` and `catch` method -- does it actually run like this?

Reply:
Oh! I see a piece of confusion on my end.
- I had told you to use `getServiceForDoc`. This is sync and in fact returns the service instead of Promise<service>. Normally this would be a bad idea if using a service from a different extension but because its within the same one that registers it, I think this is safe
- If sticking with sync then you need to update the jsdoc of `assistjsConfigServiceForDoc`'s `@return` annotation

Reply:
Done updated JSdoc. Yeah I was stuck a bit about why any async methods like `then` and `catch` don't work and found out it's actually an synchronous method.